modules/api_reference/src/Plugin/Reference/OpenApi.php

```diff
@@ -195,4 +197,51 @@ public function validate(\stdClass $content) {
     }
   }
 
+  /**
+   * Fixes values in the 'pattern' key in an OpenApi structure.
+   *
+   * @param string $key
+   *   Key of the current value, empty string for root.
+   * @param mixed $val
+   *   Current value.
+   */
+  private function fixPatterns(string $key, &$val): void {
+    if (is_object($val) || is_array($val)) {
+      foreach ($val as $k => &$v) {
+        $this->fixPatterns($k, $v);
+      }
+    }
+    elseif ($key === 'pattern') {
+      $val = static::fixRegex($val);
+    }
+  }
+
+  /**
+   * Fixes the regex so json-schema can validate it.
+   *
+   * Currently all it does is escape `/` that is not escaped and not the first
+   * character of the regex.
+   *
+   * @param string $str
+   *   Input regex string.
+   *
+   * @return string
+   *   Fixed regex.
+   */
+  private static function fixRegex(string $str): string {
```

Comment:
`private` and `static` does not make too much sense together.

Reply:
Why?

Reply:
Private something belongs to an object instance.
Static something belongs to a class.
With a (public) static method you can call on operation on a class without initiating an object from it. With a static property, you can share data between instances of the same class.

Here wherever you call `fixRegex()` you already have an object instance ($this) that you can use to call this method. Because it is private, nothing else could call it therefore `private static` does not make sense to me.